steps/24_skpro_polars_support_integration/step.md

```diff
@@ -0,0 +1,414 @@
+│ 10        ┆ 74.845273  │
+│ 0         ┆ 231.852453 │
+│ 35        ┆ 140.783099 │
+│ …         ┆ …          │
+│ 39        ┆ 149.267228 │
+│ 40        ┆ 115.727004 │
+│ 16        ┆ 252.940486 │
+│ 44        ┆ 251.146038 │
+│ 45        ┆ 137.555421 │
+└───────────┴────────────┘
+```
+
+and for frames that require melting:
+
+```python
+┌───────────────┬────────────────────────┬────────────────────────┐
+│ ____index____ ┆ __target__0.9__lower__ ┆ __target__0.9__upper__ │
+│ ---           ┆ ---                    ┆ ---                    │
+│ i64           ┆ f64                    ┆ f64                    │
+╞═══════════════╪════════════════════════╪════════════════════════╡
+│ 4             ┆ 66.772658              ┆ 176.318973             │
+│ 63            ┆ 22.517743              ┆ 132.064058             │
+│ 10            ┆ 20.072116              ┆ 129.618431             │
+│ 0             ┆ 177.079295             ┆ 286.62561              │
+│ 35            ┆ 86.009941              ┆ 195.556256             │
+│ …             ┆ …                      ┆ …                      │
+│ 39            ┆ 94.49407               ┆ 204.040385             │
+│ 40            ┆ 60.953847              ┆ 170.500162             │
+│ 16            ┆ 198.167328             ┆ 307.713643             │
+│ 44            ┆ 196.372881             ┆ 305.919196             │
+│ 45            ┆ 82.782263              ┆ 192.328578             │
+└───────────────┴────────────────────────┴────────────────────────┘
+```
+
+###### Polars to Pandas #TODO technical implementation
+
+Requires a conversion during the input check from polars to pandas. If this is not handled already in skpro, we can potentially enhance the current functionality within `_check_X` and `_convert` to check what format the input is in and then change it if necessary.
+
+~~Question 3.1~~) is this already handled? in the scenario where a polars dataframe is passed and `skpro` automatically converts it into a pandas DataFrame to calculate the predictions?
+
+~~Question 3.2~~) What should we do with the `predict` function. Currently it automatically converts whatever is passed into the `predict` function back into the mtype that was seen in fit. Do we need to refactor this as well?
+
+Question 3.3) Currently there is no way to allow the user to specify whether they want or do not want to pass back the index when converting from pandas to polars. The user currently only can specify what transform they want via `set_output` and there is no other parameter they can set to specify what configuration they want as the internal code handles the rest of the conversion. How should we tackle this problem?
+
+Question 3.4) Is there a way to check via the tags or configs as to which data container the interface is developed in regards to the private `_predict_*` methods? As an example, the `GLMRegressor` has tag 'coded_in_pandas' so we know that all the private `_predict_*` functions are written using pandas
```

Comment:
yes, the tag is `X_inner_mtype` or `y_inner_mtype`, see https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.registry._tags.x_inner_mtype.html

Currently, most implementations assume that there is only a single internal type, or multiple related ones (e.g., multiple scitypes but all pandas based).

Compositors however may assume they support all mtypes, if they only carry out abstract operations such sa passing on.